No code generation when building with .NET 6 SDK

#### Bug description
Title only

#### Repro steps
Run Build.ps1 of this MR  
https://github.com/Y-YoL/messagepack-test/pull/2

#### Expected behavior
Build succeeded.

#### Actual behavior
I get warning CS8784 and error CS0103
```
CSC : warning CS8784: Generator 'MessagePackGenerator' failed to initialize. It will not contribute to the output and c
ompilation errors may occur as a result. Exception was of type 'FileNotFoundException' with message 'Could not load fil
e or assembly 'System.Collections.Immutable, Version=7.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. 指定さ
れたファイルが見つかりません。'

error CS0103: The name 'GeneratedMessagePackResolver' does not exist in the current context 
```

- Version used: v2.6.100-alpha
- Runtime: (e.g. .NET Framework, .NET Core, Unity, mono, etc.)
```
.NET SDK (reflecting any global.json):
 Version:   6.0.408
 Commit:    0c3669d367

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22621
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.408\

Host:
  Version:      7.0.5
  Architecture: x64
  Commit:       8042d61b17
```

#### Additional context
I didn't have the issue when building using the .NET 7 SDK

--- a/Directory.Packages.props
+++ b/Directory.Packages.props
@@ -8,7 +8,7 @@
 
     <!-- https://learn.microsoft.com/en-us/visualstudio/extensibility/roslyn-version-support?view=vs-2022 -->
     <MicrosoftCodeAnalysisVersion>4.3.0</MicrosoftCodeAnalysisVersion>
-    <MicrosoftCodeAnalysisVersion Condition="'$(IsTestProject)'=='true'">4.3.0</MicrosoftCodeAnalysisVersion>
+    <MicrosoftCodeAnalysisVersion Condition="'$(IsAnalyzerProject)'!='true'">4.3.0</MicrosoftCodeAnalysisVersion>
     <MicrosoftCodeAnalysisTestingVersion>1.1.2-beta1.23163.2</MicrosoftCodeAnalysisTestingVersion>
 
     <!-- Unity 2021.3 has Roslyn 3.8.0 (https://docs.unity3d.com/Manual/roslyn-analyzers.html) -->
@@ -26,7 +26,7 @@
     <PackageVersion Include="Jil" version="2.17.0" />
     <PackageVersion Include="MessagePack" Version="2.1.90" />
     <PackageVersion Include="Microsoft.AspNetCore.Mvc.Core" Version="2.2.5" />
-    <PackageVersion Include="Microsoft.Bcl.AsyncInterfaces" Version="6.0.0" />
+    <PackageVersion Include="Microsoft.Bcl.AsyncInterfaces" Version="7.0.0" />
     <PackageVersion Include="Microsoft.Build.Framework" Version="16.5.0" ExcludeAssets="runtime" />
     <PackageVersion Include="Microsoft.Build.Locator" Version="1.5.5" />
     <PackageVersion Include="Microsoft.Build.Utilities.Core" Version="16.5.0" />
@@ -74,7 +74,16 @@
     <PackageVersion Include="xunit" Version="2.4.2" />
     <PackageVersion Include="ZeroFormatter" Version="1.6.4" />
   </ItemGroup>
+  <ItemGroup Condition="'$(IsAnalyzerProject)'=='true'">
+    <PackageVersion Update="Microsoft.Bcl.AsyncInterfaces" Version="6.0.0" />
+    <PackageVersion Update="System.Collections.Immutable" Version="6.0.0" />
+    <PackageVersion Update="System.Memory" Version="4.5.4" />
+    <PackageVersion Update="System.Reflection.Metadata" Version="6.0.0" />
+    <PackageVersion Update="System.Runtime.CompilerServices.Unsafe" Version="6.0.0" />
+    <PackageVersion Update="System.Text.Json" Version="6.0.0" />
+  </ItemGroup>
   <ItemGroup Condition="'$(MicrosoftCodeAnalysisVersion)'=='$(CodeAnalysisVersionForUnity)'">
+    <!-- These versions carefully chosen to support Unity. -->
     <PackageVersion Update="System.CodeDom" Version="5.0.0" />
     <PackageVersion Update="System.Collections.Immutable" Version="5.0.0" />
     <PackageVersion Update="System.Text.Json" Version="5.0.0" />

--- a/sandbox/PerfNetFramework/README.md
+++ b/sandbox/PerfNetFramework/README.md
@@ -10,7 +10,7 @@ When collecting ETL traces, use these settings in the Collect->Run dialog:
 
 | Setting     | Value |
 |-------------|-------|
-| Command     | `dotnet run -c release -p .\sandbox\PerfNetFramework\ -f net461 --no-build`
+| Command     | `dotnet run -c release -p .\sandbox\PerfNetFramework\ -f net472 --no-build`
 | Current Dir | `d:\git\messagepack-csharp` (or wherever your enlistment is)
 | Additional Providers | `*MessagePack-Benchmark`
 | No V3.X NGen | Checked

--- a/sandbox/TestData.InvalidProject/TestData.InvalidProject.csproj
+++ b/sandbox/TestData.InvalidProject/TestData.InvalidProject.csproj
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>net461</TargetFramework>
+    <TargetFramework>net462</TargetFramework>
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="MessagePack" />

--- a/sandbox/TestData.InvalidSyntax/TestData.InvalidSyntax.csproj
+++ b/sandbox/TestData.InvalidSyntax/TestData.InvalidSyntax.csproj
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>net461</TargetFramework>
+    <TargetFramework>net462</TargetFramework>
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="MessagePack" />

--- a/sandbox/TestData2/TestData2.csproj
+++ b/sandbox/TestData2/TestData2.csproj
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>net461</TargetFramework>
+    <TargetFramework>net462</TargetFramework>
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="MessagePack" />

--- a/src/Directory.Build.props
+++ b/src/Directory.Build.props
@@ -3,6 +3,7 @@
     <Nullable>enable</Nullable>
     <SignAssembly>True</SignAssembly>
     <AssemblyOriginatorKeyFile>$(MSBuildThisFileDirectory)..\opensource.snk</AssemblyOriginatorKeyFile>
+    <EnforceExtendedAnalyzerRules Condition="'$(IsAnalyzerProject)'=='true'">true</EnforceExtendedAnalyzerRules>
   </PropertyGroup>
   <Import Project="$([MSBuild]::GetPathOfFileAbove($(MSBuildThisFile), $(MSBuildThisFileDirectory)..))" />
 </Project>

--- a/src/MessagePack.Analyzers.CodeFixes/Directory.Build.props
+++ b/src/MessagePack.Analyzers.CodeFixes/Directory.Build.props
@@ -1,0 +1,6 @@
+<Project>
+  <PropertyGroup>
+    <IsAnalyzerProject>true</IsAnalyzerProject>
+  </PropertyGroup>
+  <Import Project="$([MSBuild]::GetPathOfFileAbove($(MSBuildThisFile), $(MSBuildThisFileDirectory)..))" />
+</Project>

--- a/src/MessagePack.Analyzers/Directory.Build.props
+++ b/src/MessagePack.Analyzers/Directory.Build.props
@@ -1,0 +1,6 @@
+<Project>
+  <PropertyGroup>
+    <IsAnalyzerProject>true</IsAnalyzerProject>
+  </PropertyGroup>
+  <Import Project="$([MSBuild]::GetPathOfFileAbove($(MSBuildThisFile), $(MSBuildThisFileDirectory)..))" />
+</Project>

--- a/src/MessagePack.SourceGenerator.Unity/Directory.Build.props
+++ b/src/MessagePack.SourceGenerator.Unity/Directory.Build.props
@@ -1,0 +1,6 @@
+<Project>
+  <PropertyGroup>
+    <IsAnalyzerProject>true</IsAnalyzerProject>
+  </PropertyGroup>
+  <Import Project="$([MSBuild]::GetPathOfFileAbove($(MSBuildThisFile), $(MSBuildThisFileDirectory)..))" />
+</Project>

--- a/src/MessagePack.SourceGenerator/Directory.Build.props
+++ b/src/MessagePack.SourceGenerator/Directory.Build.props
@@ -1,0 +1,6 @@
+<Project>
+  <PropertyGroup>
+    <IsAnalyzerProject>true</IsAnalyzerProject>
+  </PropertyGroup>
+  <Import Project="$([MSBuild]::GetPathOfFileAbove($(MSBuildThisFile), $(MSBuildThisFileDirectory)..))" />
+</Project>

